### PR TITLE
only pull skinny manifest once

### LINKED ISSUE #1101: ch-image: use the skinny manifest we got when requesting a fat one
When requesting a fat manifest, if there is not one, a skinny manifest is returned. However in this case, we immediately re-request the skinny one. We should instead use the skinny manifest we already got, possibly provided it's a v2 manifest.

### PR DESCRIPTION
**Pull request by lucaudill**
closes #1101

### CODE REVIEW

File: lib/pull.py [hunk 1]
```diff
@@ -173,9 +173,11 @@ def fatman_load(self):
                                    "manifest list: downloading")
       fm = self.fatman_path.json_from_file("fat manifest")
       if ("layers" in fm or "fsLayers" in fm):
-         # FIXME (issue #1101): If it's a v2 manifest we could use it instead
-         # of re-requesting later. Maybe we could here move/copy it over to
-         # the skinny manifest path.
+         if (not os.path.exists(str(self.manifest_path))):
+            # check for broken symlink
+            if (os.path.lexists(str(self.manifest_path))):
+               os.remove(str(self.manifest_path))
```
Comment (reidpr) [inline lib/pull.py:179]: 1. You don't actually have to worry about the dangling symlink case. Just use `lexists()` one time.
2. That said, is there a reason this isn't using the `Path` object methods?
3. Add a comment explaining this symlink trick.

File: test/build/55_cache.bats [hunk 2]
```diff
@@ -363,7 +363,6 @@ EOF
     [[ $status -eq 0 ]]
     [[ $output = *'. FROM'* ]]
     [[ $output = *'manifest list: downloading'* ]]
-    [[ $output = *'manifest: downloading'* ]]
```
Comment (reidpr) [inline test/build/55_cache.bats:366]: This message should still be printed for arch-aware images.

**Review by reidpr — CHANGES_REQUESTED** [review]
Nice work. See comments in-line.

File: lib/pull.py (hunk 1 above)
Comment (lucaudill) [reply lib/pull.py:179]: 1. Would you be willing to say more about this? My testing made it seem like if there is a dangling symlink present you'll get a FileExistsError. I wanted to protect against that case, even if it's unlikely.

2. No particularly strong reason, I'll consider switching.

**Review by reidpr — APPROVED** [review]
LGTM, thanks! I'll merge once we get 0.31 out the door.

**Pull request merged**